### PR TITLE
Support BetterReflection 4

### LINKED ISSUE #234: Support `roave/better-reflection` `^4.0`

### PR DESCRIPTION
**Pull request by Jean85**
Fixes #234, includes #247

### CODE REVIEW

**Comment by Ocramius** [comment]
7.3 should probably be removed from the build

**Comment by Jean85** [comment]
> 7.3 should probably be removed from the build

Why?

BTW rebased on top of green build in #247, now this should be green too.

**Comment by Ocramius** [comment]
If this is green, it can go in as-is 👍

**Comment by Ocramius** [comment]
I think we should only keep BetterReflection 4.6+ here: that's what the mutation tests seem to also pick up here (although I don't see the mutant from my phone)

**Comment by Jean85** [comment]
`--show-mutants` is missing in CI, I'll add it:
```
Escaped mutants:
================


1) /home/alai/workspace/OSS/BackwardCompatibilityCheck/src/DetectChanges/BCBreak/PropertyBased/PropertyDocumentedTypeChanged.php:39    [M] UnwrapArrayUnique

--- Original
+++ New
@@ @@
             return Changes::empty();
         }
         try {
-            $fromTypes = array_unique($fromProperty->getDocBlockTypeStrings());
+            $fromTypes = $fromProperty->getDocBlockTypeStrings();
             $toTypes = array_unique($toProperty->getDocBlockTypeStrings());
         } catch (InvalidArgumentException $failedToParseDocblock) {
             // @TODO #134 improve docblock parsing upstream to remove this generic try-catch


2) /home/alai/workspace/OSS/BackwardCompatibilityCheck/src/DetectChanges/BCBreak/PropertyBased/PropertyDocumentedTypeChanged.php:40    [M] UnwrapArrayUnique

--- Original
+++ New
@@ @@
         }
         try {
             $fromTypes = array_unique($fromProperty->getDocBlockTypeStrings());
-            $toTypes = array_unique($toProperty->getDocBlockTypeStrings());
+            $toTypes = $toProperty->getDocBlockTypeStrings();
         } catch (InvalidArgumentException $failedToParseDocblock) {
             // @TODO #134 improve docblock parsing upstream to remove this generic try-catch
             return Changes::empty();
```

I'll try to fix them...

File: infection.json.dist [hunk 1]
```diff
@@ -11,6 +11,12 @@
     "mutators": {
         "@default": true,
         "IdenticalEqual": false,
-        "NotIdenticalNotEqual": false
+        "NotIdenticalNotEqual": false,
+        "UnwrapArrayUnique": {
```
Comment (Ocramius) [inline infection.json.dist:15]: These should likely have tests, not be unwrapped...

**Comment by Ocramius** [comment]
>  It seems that the uniqueness applied to those arrays is already done by BetterReflection, is it right? Is it possible to ignore those mutants?

If the uniqueness isn't needed, then it should likely be removed: mutation testing works both ways, finding unused code or untested code :-)

**Comment by Jean85** [comment]
Ok :+1: Since I don't know at which point that mutant will start appearing, I bumped BetterReflection to 4.3, which seems to be the latest minor supporting 7.3, and it doesn't mutate.

**Comment by Ocramius** [comment]
Excellent! \o/

Thanks @Jean85, you basically did what I was supposed to do :D

**Pull request merged**